### PR TITLE
Start using modules

### PR DESCRIPTION
**Pull request by Didainius**
Introducing go modules as per #162.

**Note** This PR goes together with https://github.com/vmware/go-vcloud-director/pull/163 and ~~should be merged~~ is merged after go-vcloud-director is done.

~~**Note** This PR is done so that it is possible to test and includes a `replace` directive in `go.mod` which points to my forked `go-vcloud-director` repo. Once https://github.com/vmware/go-vcloud-director/pull/163 is merged it _**must be adjusted.**_~~

I have intentionally made separate commits so that it is easier to review because enabling modules messes with vendor directory which has a lot of files and makes review inconvenient.

### Drill down
The general idea behind it is such:
* Enable modules and set major version to be `v2`. Initial go.mod dependencies rebuilt from vendor/vendor.json   (fc907b0)
* Repoint local imports to use `v2` version of package so that it doesn't search for `v0\v1` (1edfd63)
* Repoint imports of `go-vcloud-director` to import `v2` package.  (0173742)
~~* Use `replace` directive before https://github.com/vmware/go-vcloud-director/pull/163 is merged to use `go-vcloud-director` from my forked repo (d819cf3)~~
* Remove the old vendor directory (179b7cc)
* Do `go mod tidy` to cleanup unused imports (b0ca9ff)
* Rebuild `vendor` directory to make repo compliant with older go versions (58b028c)
* Configure Travis for testing both `modules` and `vendor` ways of dependencies (bb07c19). Travis checks (using Go 1.12 9cef71a ) if vendor is in sync with go.mod using `make vendor-check` when `GO111MODULE=on` .

### Testing
**Note** Do not forget to set `GO111MODULE=on` if you're in `$GOPATH` to explicitly test how modules (not vendor directory) works. It is well worth to clean your module cache between builds with `go clean -modcache`.

To test simply go somewhere outside $GOPATH or set `export GO111MODULE=on` and also test it inside gopath with `GO111MODULE=auto` or `GO111MODULE=off`

**Note** make vendor-check should complain when `GO111MODULE=off` or `GO111MODULE=auto` in `$GOPATH`
```
git clone git@github.com:Didainius/terraform-provider-vcd.git
cd terraform-provider-vcd/
git checkout start_using_modules
make vendor-check
make test
make build
make install
```

#### Tested
* In MacOS inside and outside of go modules.
* In Ubuntu Linux inside and outside GOPATH with `GO111MODULE=off`, `GO111MODULE=on` and `GO111MODULE=auto`

### Missing
~~* Documentation updates for contributors~~

### CODE REVIEW

**Comment by vbauzysvmware** [comment]
Check for formatting changes. I am sure there are some. Giuseppe showed that.

**Comment by vbauzysvmware** [comment]
Why so many changes in Vendor folder?

**Comment by Didainius** [comment]
> Check for formatting changes. I am sure there are some. Giuseppe showed that.

Surprises myself, but none reported in Go 1.10, 1.11 and 1.12.

**Comment by Didainius** [comment]
> Why so many changes in Vendor folder?

Because it is being rebuilt from scratch using modules and seems there were a lot of stale packages in /vendor before.

File: README.md [hunk 1]
```diff
@@ -32,22 +32,54 @@ $ cd $GOPATH/src/github.com/terraform-providers/terraform-provider-vcd
...
+This means that it is no longer necessary to be in GOPATH.
+[See more](https://github.com/golang/go/wiki/Modules#how-to-use-modules) on how to use modules
+and toggle between modes.
 
-To compile the provider, run `make build`. This will build the provider and put the provider binary in the `$GOPATH/bin` directory.
-
-```sh
+```
+$ cd ~/mydir
+$ clone git@github.com:terraform-providers/terraform-provider-vcd
```
Comment (dn1s) [inline README.md]: For copy + paste reasons you should write `git clone git@github.com:terraform-providers/terraform-provider-vcd`

**Pull request merged**